parsl/executors/high_throughput/process_worker_pool.py

```diff
@@ -239,23 +273,28 @@ def start(self):
         self._task_puller_thread.start()
         self._result_pusher_thread.start()
 
-        abort_flag = False
-
         logger.info("Loop start")
 
         # TODO : Add mechanism in this loop to stop the worker pool
         # This might need a multiprocessing event to signal back.
-        while not abort_flag:
+        while not self._kill_event.is_set():
             time.sleep(0.1)
+        logger.critical("[MAIN] Received kill event, terminating worker processes")
 
+        self._task_puller_thread.join()
+        self._result_pusher_thread.join()
         for proc_id in self.procs:
             self.procs[proc_id].terminate()
+            self.procs[proc_id].join()
+            logger.critical("Terminating worker {}:{}".format(self.procs[proc_id],
```

Comment:
log this before the terminate/join, so that when it takes a long time you can see which one it is?

Reply:
Done, and added a debug statement after the join.